packages/experiments/src/components/Toggle/Toggle.styles.ts

```diff
@@ -1,222 +1,229 @@
-        backgroundColor: thumbBackground,
-        /* Border is added to handle high contrast mode for Firefox */
-        borderColor: 'transparent',
-        borderWidth: '.28em',
-        borderStyle: 'solid',
-        boxSizing: 'border-box'
-      },
-      !disabled &&
-        checked && [
-          {
-            backgroundColor: thumbCheckedBackground,
+            borderWidth: '.28em',
+            borderStyle: 'solid',
+            boxSizing: 'border-box',
             selectors: {
               [HighContrastSelector]: {
-                backgroundColor: 'Window',
-                borderColor: 'Window'
+                backgroundColor: state.thumbHighContrastBackground,
+                borderColor: state.thumbHighContrastBorderColor
               }
             }
           }
         ],
-      disabled && [
-        !checked && [
-          {
-            backgroundColor: thumbDisabledBackground
-          }
-        ],
-        checked && [
+
+        text: [
+          classNames.text,
           {
-            backgroundColor: thumbCheckedDisabledBackground
-          }
-        ]
-      ]
-    ],
-
-    text: [
-      'ms-Toggle-stateText',
-      {
-        selectors: {
-          // Workaround: make rules more sepecific than Label rules.
-          '&&': {
-            padding: '0',
-            margin: '0 10px',
-            userSelect: 'none'
-          }
-        }
-      },
-      disabled && {
-        selectors: {
-          '&&': {
-            color: textDisabledColor,
             selectors: {
-              [HighContrastSelector]: {
-                color: 'GrayText'
+              // Workaround: make rules more sepecific than Label rules.
+              '&&': {
+                color: state.textColor,
+                padding: '0',
+                margin: '0 10px',
+                userSelect: 'none',
+                selectors: {
+                  [HighContrastSelector]: {
+                    color: state.textHighContrastColor
+                  }
+                }
               }
             }
           }
-        }
-      }
-    ]
-  };
+        ]
+      };
+    }
+
+    // No state
+    return {};
+  }
+
+  function getToggleStylesFromVariant(): Partial<IToggleStyles> {
+    return concatStyleSets(
+      getToggleStylesFromState(toggleVariables.baseState),
+      !disabled && !checked && getToggleStylesFromState(toggleVariables.enabled),
+      disabled && !checked && getToggleStylesFromState(toggleVariables.disabled),
+      !disabled && checked && getToggleStylesFromState(toggleVariables.checked),
+      disabled && checked && getToggleStylesFromState(toggleVariables.checkedDisabled)
+    );
+  }
+
+  return concatStyleSets({ root: getFocusStyle(theme) }, getToggleStylesFromVariant(), { root: className });
 };
+
+type IProcessedVariables<T> = { [P in keyof T]-?: IProcessedVariables<T[P]> };
+
+function _processVariables<T>(partialVariables: T, customVariables?: T): IProcessedVariables<T> {
```

Comment:
@JasonGore @dzearing

I copied this from Button.styles.ts.

I know it should probably go into Utilities as something all components will be using, but wanted your input on where exactly before moving it.

Reply:
For this PR at least it could go into `experiments/src/utilities/` to remove the duplication with Button and then finalized when one of the controls using it is promoted.